hunts/src/tagFilter.js

```diff
@@ -0,0 +1,24 @@
+function getDescendants(tableRow) {
+  const allChildren = [tableRow];
+  for (let i = 0; i < allChildren.length; i++) {
+    const current = allChildren[i];
+    Array.prototype.push.apply(allChildren, current.subRows);
+  }
+  return allChildren;
+}
+
+export function addOrRemoveTag(tagList, tag) {
+  let tagInList = tagList.filter(x => ['name', 'color', 'id'].every(key => x[key] === tag[key]))[0];
+  return (tagInList !== undefined) ? tagList.filter((i) => i !== tagInList) : tagList.concat(tag);
+}
+
+export function filterPuzzlesByTagfn(rows, id, filterValue) {
+  let requiredRowIDs = rows.filter((row) => {
+    return tagList.length === 0 || tagList.some(tag => row.original.tags.map(x => x.id).includes(tag.id));
+  }).map(row => row.id);
+  // Show metas with puzzles with tags we want? Not sure if this is a good idea.
```

Comment:
I'd advocate for keeping it simple and only match direct tags, not descendants, since I imagine users might get confused to see puzzles listed that don't directly match tags. I could be convinced otherwise though :D

Reply:
OK, will change that, hopefully it doesn't make things break. (I wasn't actually testing with a meta, might do that.)

Reply:
oh yeah, might be good to sanity check with a meta to see if there's any weird behavior

Reply:
Somehow (once I got it to work in the first place, which right now is only locally) it seems to work with a meta, not sure I understand how. Will look into that before pushing, because I want to have some idea why this code is working.

Reply:
I think I understand how it works now. The indentation is a bit weird (puzzles in metas are indented whether or not the meta is visible), I'll create an issue for that (also applies to the show all/priority/unsolved settings). But I don't think that's critical.